jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -826,9 +826,10 @@ void java_object_factoryt::gen_nondet_struct_init(
     // This code mirrors the `remove_java_new` pass:
     auto initial_object = zero_initializer(expr.type(), source_locationt(), ns);
     CHECK_RETURN(initial_object.has_value());
-    const irep_idt qualified_clsid = "java::" + id2string(struct_tag);
     set_class_identifier(
-      to_struct_expr(*initial_object), ns, struct_tag_typet(qualified_clsid));
+      to_struct_expr(*initial_object),
+      ns,
```

Comment:
:bulb: `set_class_identifier` doesn't actually use `ns`. So this parameter could probably be removed, however that is probably a change for another PR.

Reply:
It has a call to `ns.follow`. I still find it a bit strange that `follow` is defined only for namespaces and not for symbol tables, but that would definitely be a change for another (much bigger) PR. :slightly_smiling_face:

Reply:
Ah, my mistake.